src/gui/codeeditors/qgscodeeditorexpression.h

```diff
@@ -56,6 +56,11 @@ class GUI_EXPORT QgsCodeEditorExpression : public QgsCodeEditor
      */
     void setFields( const QgsFields &fields );
 
+    /**
+     * Toggles comments for selected lines.
+     */
+    void toggleComment() override;
```

Comment:
As per the docs for QgsCodeEditor::toggleComment, you'll also need to override QgsCodeEditor::languageCapabilities() and return the ToggleComment capability.